### ISSUE TITLE
Flavors: Add `afm` flavor support for Agent-Flavored Markdown (AFM)

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#402]
## Problem

rumdl flavors reduce false positives for Markdown variants. Agent-Flavored Markdown (AFM) defines agent files with:
- A required YAML front matter block (`--- ... ---`).
- A Markdown body that MUST include `# Role` and `# Instructions`.

In AFM files, this commonly triggers:
- `MD041` when the file starts with front matter instead of a heading.
- `MD025` because AFM expects multiple H1 headings (`# Role`, `# Instructions`).

## Request

Add a new flavor: `afm`, aligned with the AFM specification, to lint AFM files cleanly.

```toml
[global]
flavor = "afm"

[per-file-flavor]
"**/*.afm" = "afm"
"**/*.afm.md" = "afm"
```

### Auto-detection

When no flavor is configured, auto-detect:
- `.afm` -> `afm`
- `.afm.md` -> `afm`

## Expected behavior

Given:

```yaml
---
spec_version: "0.3.0"
name: "Example Agent"
---
# Role
...

# Instructions
...
```

- `MD041` should treat the first heading after YAML front matter as the document start.
- `MD025` should allow multiple H1 headings for AFM (at minimum, `Role` and `Instructions`).

## Acceptance criteria

- `afm` is a supported `flavor` value and is documented.
- `.afm` and `.afm.md` are auto-detected as `afm` when no flavor is set.
- Add fixtures/tests that demonstrate `MD041` and `MD025` no longer false-positive on valid AFM files.

## References

- https://wso2.github.io/agent-flavored-markdown/specification/
- https://github.com/wso2/agent-flavored-markdown/blob/main/spec/docs/specification.md

**Comment by rvben** [comment]
Thanks for filing this, @pygarap. AFM is still very early (spec v0.3.0, 16 stars, no releases), so we'll hold off on a dedicated flavor for now. The AFM format is essentially standard markdown with required headings and YAML frontmatter — the linting impact is minimal and can be handled with config:
```toml
[per-file-ignores]
"**/*.afm" = ["MD025"]
"**/*.afm.md" = ["MD025"]
```

Happy to revisit if/when AFM reaches a stable 1.0 release and gains broader adoption.
